modules/promotion/src/Plugin/Commerce/PromotionOffer/PercentageOffBase.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer;
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides base class for percentage off offers.
+ */
+abstract class PercentageOffBase extends PromotionOfferBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'amount' => 0,
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * Gets the percentage amount, as a decimal, negated.
+   *
+   * @return string
+   *   The amount.
+   */
+  public function getAmount() {
+    return (string) (-$this->configuration['amount']);
```

Comment:
Well \Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer\PromotionOfferBase::applyAdjustment already has a Price passed to it to pass to the Adjustment.

and in \Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer\ProductPercentageOff::execute we invoke Multiple. I guess in the execute we can flip \Drupal\commerce_promotion\Plugin\Commerce\PromotionOffer\PercentageOffBase::getAmount.

Just felt like better DX.